examples/example_host.cpp

```diff
@@ -21,153 +21,122 @@ struct account
-}
-
-static evmc_uint256be get_balance(evmc_context* context, const evmc_address* address)
-{
-    auto* host = static_cast<example_host_context*>(context);
-    auto it = host->accounts.find(*address);
-    if (it != host->accounts.end())
-        return it->second.balance;
-    return {};
-}
-
-static size_t get_code_size(evmc_context* context, const evmc_address* address)
-{
-    auto* host = static_cast<example_host_context*>(context);
-    auto it = host->accounts.find(*address);
-    if (it != host->accounts.end())
-        return it->second.code_size;
-    return 0;
-}
-
-static evmc_bytes32 get_code_hash(evmc_context* context, const evmc_address* address)
-{
-    auto* host = static_cast<example_host_context*>(context);
-    auto it = host->accounts.find(*address);
-    if (it != host->accounts.end())
-        return it->second.code_hash;
-    return {};
-}
-
-static size_t copy_code(evmc_context* context,
-                        const evmc_address* address,
-                        size_t code_offset,
-                        uint8_t* buffer_data,
-                        size_t buffer_size)
-{
-    (void)context;
-    (void)address;
-    (void)code_offset;
-    (void)buffer_data;
-    (void)buffer_size;
-    return 0;
-}
-
-static void selfdestruct(evmc_context* context,
-                         const evmc_address* address,
-                         const evmc_address* beneficiary)
-{
-    (void)context;
-    (void)address;
-    (void)beneficiary;
-}
-
-static evmc_result call(evmc_context* context, const evmc_message* msg)
-{
-    (void)context;
-    (void)msg;
-    evmc_result result{};
-    result.status_code = EVMC_FAILURE;
-    return result;
-}
-
-static evmc_tx_context get_tx_context(evmc_context* context)
-{
-    (void)context;
-    evmc_tx_context result{};
-    return result;
-}
-
-static evmc_bytes32 get_block_hash(evmc_context* context, int64_t number)
-{
-    auto* host = static_cast<example_host_context*>(context);
-    int64_t current_block_number = host->tx_context.block_number;
-
-    auto example_block_hash = evmc_bytes32{};
-    if (number < current_block_number && number >= current_block_number - 256)
-        example_block_hash = {{1, 1, 1, 1}};
-    return example_block_hash;
-}
-
-static void emit_log(evmc_context* context,
-                     const evmc_address* address,
-                     const uint8_t* data,
-                     size_t data_size,
-                     const evmc_bytes32 topics[],
-                     size_t topics_count)
-{
-    (void)context;
-    (void)address;
-    (void)data;
-    (void)data_size;
-    (void)topics;
-    (void)topics_count;
-}
 
-static const evmc_host_interface interface = {
-    account_exists, get_storage,  set_storage, get_balance,    get_code_size,  get_code_hash,
-    copy_code,      selfdestruct, call,        get_tx_context, get_block_hash, emit_log,
+public:
+    bool account_exists(const evmc_address& addr) noexcept override
```

Comment:
could be `final`

Reply:
Yes, I noticed. Decided to go `override` because more common. I tend to not to use less common C++ features in examples. Although, in this case it might be good suggestion to use `final` here.